app/src/main/java/mozilla/lockbox/support/SecurePreferences.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.support
+
+import android.content.SharedPreferences
+import android.util.Base64
+import mozilla.components.lib.dataprotect.Keystore
+import java.nio.charset.StandardCharsets
+
+private const val KEYSTORE_LABEL = "lockbox-keystore"
+private const val BASE_64_FLAGS = Base64.URL_SAFE or Base64.NO_PADDING
+
+open class SecurePreferences(
```

Comment:
At some point, `Keystore.generateKey()` must be called, or all the calls to encrypt/decrypt will error.

Reply:
oh.

Reply:
my slight bias is to do this lazily, at the point where encrypt/decrypt are first called:

* the provisioning is cheap, so the latency is minor
* it's arguably a better place in the app's lifecycle to deal with errors?
* If it doesn't exist when there's definitely a value to decrypt, we can differentiate the failure better (e.g., _something_ about the app left `SharedPreferences` around but not an encryption key, vs the decrypt failed because the keys are different)?

Reply:
Ok, I'd like to link up with you to talk through the error cases and what handling we should have here.